### PR TITLE
Make journey-test up to date with exercism cli updates

### PR DESCRIPTION
**Pull request by FridaTveit**
The exercism cli has been updated, therefore we need to change our build script so that it still works.

Problems we still need to fix:
- `exercism configure` can't be called without a token so I'm currently using my own token. We should find out what the standard procedure is for this, e.g. creating a token for the track or using some option that allows us not to use a token.

- I get the error `The base API URL 'http://localhost:9292' cannot be reached` even though xapi should be running. Apparently other tracks have had this issue as well.

- `exercism --config` doesn't appear to be a valid option anymore. I can't find the config file we were using so I'm hoping it's fine to not use it anymore?

See issue #1493 for details.

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

File: bin/journey-test.sh [hunk 1]
```diff
@@ -305,7 +304,6 @@ main() {
   local trackler_home="${build_path}/trackler"
   local exercism_home="${build_path}/exercism"
 
-  local exercism_configfile=".journey-test.exercism.json"
```
Comment (FridaTveit) [inline bin/journey-test.sh]: Does anyone know anything about this file? I can't find it anywhere so I'm not sure if it's important or not...

**Comment by FridaTveit** [comment]
Hm, the build passed... That's unexpected...

I think it's because `run-journey-test-from-ci` doesn't detect any changes to the exercises or `config.json` and therefore doesn't run `journey-test`. We probably want it to solve all exercises if any of the scripts have changed or if `config.json` or any `.gradle` file has changed?

**Comment by FridaTveit** [comment]
I'm working on a change to `run-journey-test-from-ci` to fix this. I think it works, just need to test it a bit more! Will push it to this PR when I'm done 🙂

**Comment by FridaTveit** [comment]
Done 🙂 I did a lot of testing (though of course I might have missed something) and I think the script works now 🙂 I've assumed that we want to run all exercises if a `.gradle` file or a `.sh` file or `config.json` has changed, let me know if anyone disagrees with that 🙂

**Comment by sjwarner-bp** [comment]
Woohoo! Thanks for that @FridaTveit - I'll take a closer look when I get a chance.

Just as a quick update, Katrina has left the message:

> We have an open issue for figuring out a more robust set of tests for tracks. The journey tests are (as you discovered) not going to work with the new CLI directly. I'm sorry I forgot to think about this during the rush of the launch!

It's really great to have those changes in to allow us to run Travis on those changes too :tada:

**Comment by FridaTveit** [comment]
I see, thanks for the update! 🙂

**Comment by FridaTveit** [comment]
Given that there's not been much progress on https://github.com/exercism/exercism/issues/3955 the best temporary solution might be to just iterate over the exercises on `config.json` and not use the exercism cli. As long as we don't squash on merge for this PR we'll still have the nearly working commit to go back to if we decide to use exercism cli again. What do you think @sjwarner-bp, @exercism/java? 🙂

**Comment by Smarticles101** [comment]
@FridaTveit that sounds like a good solution to me, even if it's just temporary.  What value does using the cli add exactly rather than just iterating over the exercises in `config.json` exactly?

Sorry I haven't been active recently

**Comment by sjwarner-bp** [comment]
@FridaTveit I think that sounds like a good temp solution :slightly_smiling_face: 

Until there is a nice way to use the new CLI, I'm all for steering clear of it :slightly_smiling_face:

**Comment by FridaTveit** [comment]
Thanks, you too! 🙂

**Comment by FridaTveit** [comment]
This is ready to be reviewed now 🙂 

@Smarticles101, sorry I didn't see your comment earlier! Using the cli meant that the tests were more similar to what the users will do so should allow us to spot more issues. But I can't think of any examples in particular of issues I think using the cli would find that our now more traditional test runner wouldn't. Apart from spotting if an exercise doesn't have a uuid since that would be required to use the cli. But hopefully we would spot that during the review anyway 🙂

**Pull request merged**